### PR TITLE
Module Dependencies and APM files for InRobot

### PR DESCRIPTION
**Pull request by ronpal**
# Description

Bringing back files from the presumed-to-be obsolete `cdf_apm_base` module.

Support for module dependencies. If a module is a dependency, it is automatically included. 

Screenshot 2024-10-07 at 21 07 17

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by ana-gheorghe — APPROVED** [review]

**Comment by ana-gheorghe** [comment]
Looks good I think. Do we need to specify somewhere that choosing InRobot setup requires this module in addition to the inrobot one?

**Comment by ronpal** [comment]
> Looks good I think. Do we need to specify somewhere that choosing InRobot setup requires this module in addition to the inrobot one?

Kinda, but we will implement a dependency in the code so no manual intervention is necessary

**Review by ana-gheorghe — APPROVED** [review]

File: cognite_toolkit/_cdf_tk/commands/modules.py [hunk 1]
```diff
@@ -256,6 +256,35 @@ def init(
 
         raise typer.Exit()
 
+    def _select_modules_in_package(self, package: Package) -> list[ModuleLocation]:
+        choices = []
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/modules.py:260]: Unused
```suggestion
```

File: cognite_toolkit/_cdf_tk/commands/modules.py [hunk 2]
```diff
@@ -256,6 +256,35 @@ def init(
...
+        dependencies: list[str] = []
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/modules.py:261]: There could be repeating dependencies so lets use a set
```suggestion
        dependencies: set[str] = set()
```

File: tests/test_unit/test_cli/test_build_deploy_snapshots/cdf_apm_base.yaml [hunk 3]
```diff
@@ -1,708 +1 @@
...
-    title:
-      container:
-        externalId: APM_Operation
-        space: APM_SourceData
-        type: container
-      containerPropertyIdentifier: title
-      description: Brief title or summary of the specified operation.
-  space: APM_SourceData
-  version: '1'
+{}
```
Comment (doctrino) [inline tests/test_unit/test_cli/test_build_deploy_snapshots/cdf_apm_base.yaml:1]: Something looks wrong here

File: cognite_toolkit/_builtin_modules/inrobot/cdf_inrobot_location/module.toml [hunk 4]
```diff
@@ -1,5 +1,6 @@
 [module]
 title = "InRobot Location"
+default_selected = true
```
Comment (doctrino) [inline cognite_toolkit/_builtin_modules/inrobot/cdf_inrobot_location/module.toml:3]: Suggest to call this 
```suggestion
is_selected_by_default = true
```

File: cognite_toolkit/_builtin_modules/inrobot/cdf_inrobot_common/module.toml [hunk 5]
```diff
@@ -1,7 +1,11 @@
...
 title = "InRobot Common"
+default_selected = true
 
 [packages]
 tags = [
     "inrobot",
 ]
+
+[dependencies]
+modules = ["cdf_apm_base"]
```
Comment (doctrino) [inline cognite_toolkit/_builtin_modules/inrobot/cdf_inrobot_common/module.toml:11]: Very explicit, but I think that is good.

Just to show an alternative, which I don't thin is necessarily better.

```yaml
[module]
...
dependencies = ["cdf_apm_base"]

```

**Review by doctrino — COMMENTED** [review]
Overall, I like it a lot. I think `is_selected_by_default` is clearer. The rest of the comments are more nitpicking.

**Review by doctrino — APPROVED** [review]

**Pull request merged**